### PR TITLE
Register Member Page with functionality Done

### PR DESCRIPTION
**Pull request by DinolanNaidoo**

### CODE REVIEW

File: skietbaan_fe/src/components/RegisterMember.js [hunk 1]
```diff
@@ -36,21 +42,54 @@ class App extends Component {
...
   }
 
+  UpdateMember() {
+    let RequestObject = {
+      "username": document.getElementById("usernameValue").value,
+      "memberID": document.getElementById("membershipID").value,
+      "entryDate": document.getElementById("entrydate").value + "T00:00:00",
+      "memberExpiry": document.getElementById("expdate").value + "T00:00:00"
+    }
+       fetch("http://api.skietbaan.retrotest.co.za/api/Features/Update" , {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/RegisterMember.js]: Aren't your api calls supposed to be a in a service.js file ? Again the url should be in a const file

File: skietbaan_fe/src/components/RegisterMember.js [hunk 2]
```diff
@@ -36,21 +42,54 @@ class App extends Component {
...
+       method: 'Put',	
+         headers: {'Accept': 'application/json','Content-Type': 'application/json'},
+         body: JSON.stringify(RequestObject)
+       }).then(function(response) {
+          ToastStore.success(document.getElementById("usernameValue").value + " is successfully registered as a member ");
+
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/RegisterMember.js]: Clean up the empty lines and fix indentations

File: skietbaan_fe/src/components/RegisterMember.js [hunk 3]
```diff
@@ -36,21 +42,54 @@ class App extends Component {
...
-  RegisterMember() {
+  SearchMember() {
    if(this.state.validForm)
    {
-    var RequestObject = {
-        "Username":this.state.usernameValue,
-    }
-      const BASE_URL = 'http://skietbaan.retrotest.co.za/';
-      fetch(`${BASE_URL}`,"/api/values", {
+      fetch("http://api.skietbaan.retrotest.co.za/api/Features/Search?Username="+this.state.usernameValue, {
```
[the remark this answers is not in the document]
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/RegisterMember.js]: we had issues with that one so i changed it back to the full url. It was not picking up the the const variable

File: skietbaan_fe/src/components/RegisterMember.js [hunk 4]
```diff
@@ -36,21 +42,54 @@ class App extends Component {
...
       method: 'Get',	
         headers: {'Accept': 'application/json','Content-Type': 'application/json'},
-        body: JSON.stringify(RequestObject)}).then(function(response) {
-          return response.json();}).then(function(data) {}).catch(function() {});
+      }).then(function(response) {
+          return response.json();})
+          .then(function(data) {
+            if(data.memberID == null){
+              document.getElementById("membershipID").value = data.memberID;
```
[the remark this answers is not in the document]
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/RegisterMember.js]: I have removed the if statement

File: skietbaan_fe/src/components/RegisterMember.js (hunk 1 above)
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/RegisterMember.js]: the url wan not being read when i used the const variable

File: skietbaan_fe/src/components/RegisterMember.js (hunk 2 above)
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/RegisterMember.js]: i have done that

File: skietbaan_fe/src/components/RegisterMember.js [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import React, { Component } from 'react';
-import { Container, Col, Form, FormGroup, Label, Input, Button, } from 'reactstrap';
+import { Container, Col, Form, FormGroup, Label, Input, Button } from 'reactstrap';
 import '../components/RegisterMemberStyles.css';
+import {ToastContainer, ToastStore} from 'react-toasts';
 
  function validateUsername(username) {
   var re = /[a-zA-Z]/;
```
[the remark this answers is not in the document]
Comment (DinolanNaidoo) [reply skietbaan_fe/src/components/RegisterMember.js]: i have made the change

**Pull request merged**